app/models/ems_refresh/save_inventory.rb

```diff
@@ -105,7 +107,7 @@ def save_vms_inventory(ems, hashes, target = nil)
             h[:location] = "unknown" if h[:location].blank?
 
             # build a type-specific vm or template
-            found = ems.vms_and_templates.build(h)
+            found = ems.vms_and_templates.klass.new(h)
```

Comment:
@Ladas yea, I think there is a bug in rails here.
Glad we came up with an easy fix but...
would be nice to fix it at the source.

I can't believe you found this issue though. very cool